Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -120,7 +120,7 @@ extension Processor {
   mutating func advance(to nextIndex: Input.Index) {
     assert(nextIndex >= bounds.lowerBound)
     assert(nextIndex <= bounds.upperBound)
-    assert(nextIndex > currentPosition)
+    assert(nextIndex >= currentPosition)
```

Comment:
Is this the better change, or is it better to not call `advance(to:)` if there's nowhere to advance to?

Reply:
It doesn't seem necessary to require the precondition for this method, given that a zero-width "advance" is a reasonable output for a consumer/other regex component. Would it sound better to you to rename this to `continue(at:)`?